src/o365/commands/accesstoken-get.ts

```diff
@@ -0,0 +1,88 @@
+import commands from './commands';
+import GlobalOptions from '../../GlobalOptions';
+import Command, {
+  CommandOption,
+  CommandValidate,
+  CommandError
+} from '../../Command';
+import auth from '../../Auth';
+
+const vorpal: Vorpal = require('../../vorpal-init');
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  new?: boolean;
+  resource: string;
+}
+
+class AccessTokenGetCommand extends Command {
+  public get name(): string {
+    return `${commands.ACCESSTOKEN_GET}`;
+  }
+
+  public get description(): string {
+    return 'Gets access token for the specified resource';
+  }
+
+  public commandAction(cmd: CommandInstance, args: CommandArgs, cb: (err?: any) => void): void {
+    auth
+      .ensureAccessToken(args.options.resource, cmd, this.debug, args.options.new)
+      .then((accessToken: string): void => {
+        cmd.log(accessToken);
+        cb();
+      }, (err: any): void => cb(new CommandError(err)));
+  }
+
+  public options(): CommandOption[] {
+    const options: CommandOption[] = [
+      {
+        option: '-r, --resource <resource>',
+        description: 'The resource for which to retrieve an access token'
+      },
+      {
+        option: '--new',
```

Comment:
Rookie issue found! Should new be optional or mandatory like `--new [new]`?

Reply:
Nope, because it's a flag. When you specify `--new [new]` then you expect a value to be passed along like `--new ohyeah`. When you specify only `--new`, then it's a flag. See also flags like debug or verbose: https://github.com/pnp/office365-cli/blob/cd657ed0fbdd5bec319d846f2dce906390f28dc2/src/Command.ts#L154